src/hex_api.erl

```diff
@@ -0,0 +1,151 @@
+-module(hex_api).
+-export([
+    default_options/0,
+    get/1,
+    get/2,
+    get_package/1,
+    get_package/2,
+    get_release/2,
+    get_release/3,
+    get_user/1,
+    get_user/2
+]).
+
+-type options() :: [{client, hex_http:client()} | {uri, binary()}].
+
+%% @doc
+%% Default options used to interact with the API.
+%% @end
+-spec default_options() -> options().
+default_options() ->
+    Client = #{adapter => hex_http_httpc, user_agent_fragment => <<"(httpc)">>},
+    URI = <<"https://hex.pm/api">>,
+    [{client, Client}, {uri, URI}].
+
+-spec get(binary()) -> {ok, term()} | {error, term()}.
+get(Path) when is_binary(Path) ->
+    get(Path, default_options()).
+
+-spec get(binary(), options()) -> {ok, term()} | {error, term()}.
+get(Path, Options) when is_binary(Path) and is_list(Options) ->
+    Client = proplists:get_value(client, Options),
+    URI = proplists:get_value(uri, Options),
+    Headers = #{<<"accept">> => <<"application/vnd.hex+erlang">>},
+
+    case hex_http:get(Client, <<URI/binary, Path/binary>>, Headers) of
+        {ok, {200, _, Body}} ->
+            {ok, binary_to_term(Body)};
+
+        {ok, {404, _, _Body}} ->
+            {error, not_found};
+
+        Other ->
+            Other
+    end.
+
+%% @doc
+%% Gets package.
+%% @end
+-spec get_package(binary()) -> {ok, map()} | {error, term()}.
+get_package(Name) when is_binary(Name) ->
+    get_package(Name, []).
+
+-spec get_package(binary(), options()) -> {ok, map()} | {error, term()}.
+get_package(Name, Options) when is_binary(Name) and is_list(Options) ->
+    Fields = [
+        name,
+        repository,
+        url,
+        docs_html_url,
+        html_url,
+        {releases, [url, version]},
+        retirements,
+        {meta, [maintainers, links, licenses, description]},
+        {owners, [username, url, email]},
+        {downloads, [week, recent, day, all]},
+        inserted_at,
+        updated_at
+    ],
```

Comment:
Are the fields for string to atom conversion? I wonder if we should use strings instead so that we don't have to update this library for each new field the API adds.

Reply:
https://github.com/hexpm/hex_erl/pull/11#discussion_r177895427 :) I wanted atom keys and chose whitelist approach as opposed to blacklist approach for 2 reasons: it nicely handles deprecated fields and different shapes: downloads shape is known but links is not etc.

I am aware of increased maintenance burden of adding new fields. Hope its not too pretentious but I consider Hex to be mostly complete :) so I dont think this will be a huge issue in practice but its definitely not ideal.

If we stick to it I gotta feeling I will regres it the next time we do happen to add a new field, but let me sit on it for a couple days anyway :)

Reply:
Thanks for the reminder 😀. Sounds good.

Reply:
I've decided to revert these changes. The reason is when hex_erl is released and used by build tools, when we add a new field in hexpm we'll need to update hex_erl and then bump both hex & rebar which seems too much.

I'd still like to use atom keys here (and in tarball metadata!) but maybe will use blacklist approach later; keeping as strings for now.

sorry for going back'n'forth 😅